pkg/storaged/client.js

```diff
@@ -900,21 +903,122 @@ client.vdo_overlay = vdo_overlay();
+                    return client.stratis_manager.UnlockPool(uuid, "keyring");
+                };
+
+                client.stratis_create_pool = (name, devs, key_desc) => {
+                    return client.stratis_manager.CreatePool(name, [false, 0],
+                                                             devs,
+                                                             key_desc ? [true, key_desc] : [false, ""],
+                                                             [false, ["", ""]]);
+                };
+
+                client.stratis_list_keys = () => {
+                    return client.stratis_manager.ListKeys();
+                };
+
+                client.stratis_create_filesystem = (pool, name) => {
+                    return pool.CreateFilesystems([[name, [false, ""]]]);
+                };
+
+                client.features.stratis_found = true;
+                client.features.stratis = true;
+                client.stratis_pools = client.stratis_manager.client.proxies("org.storage.stratis3.pool.r0",
+                                                                             "/org/storage/stratis3",
+                                                                             { watch: false });
+                client.stratis_blockdevs = client.stratis_manager.client.proxies("org.storage.stratis3.blockdev.r0",
+                                                                                 "/org/storage/stratis3",
+                                                                                 { watch: false });
+                client.stratis_filesystems = client.stratis_manager.client.proxies("org.storage.stratis3.filesystem.r0",
+                                                                                   "/org/storage/stratis3",
+                                                                                   { watch: false });
+
+                return stratis.watch({ path_namespace: "/org/storage/stratis3" }).then(() => {
+                    client.stratis_manager.client.addEventListener('notify', (event, data) => {
+                        client.update();
+                    });
+                });
+            });
+}
+
+function stratis2_start() {
     const stratis = cockpit.dbus("org.storage.stratis2", { superuser: "try" });
     client.stratis_manager = stratis.proxy("org.storage.stratis2.Manager.r1",
                                            "/org/storage/stratis2");
 
+    // The rest of the code expects these to be initialized even if no
+    // stratisd is found.
     client.stratis_pools = { };
     client.stratis_blockdevs = { };
     client.stratis_filesystems = { };
-    client.stratis_manager.data.LockedPoolsWithDevs = [];
-    client.stratis_manager.data.KeyList = [];
+    client.stratis_manager.LockedPools = {};
 
     return client.stratis_manager.wait()
             .then(() => {
+                client.features.stratis_found = true;
```

Comment:
You may want to move this a bit below, where it sets `client.features.stratis = true`. That matches the stratis3 variant then. I was rather confused why this method doesn't set the flag.

Reply:
stratis3 and stratis2 are not the same in this regard.  Stratis2 might set `feature.stratis_found` to true but never set `feature.stratis` (when the version check fails).

We don't really need `stratis_found` at all, probably. It only matters when people install a stratisd version < 2.4 on a OS where we do on-demand installation of Stratis. All these OSes (fedora and centos) have stratisd 2.4.2, so one really has to get creative to cause `stratis_found` to make a difference.

It was important in #16525, where it would make life sane for people that have Stratis 3 installed.

But it seems to be too complex now, so I'll remove it, ok?

Reply:
Done.